backend/src/main/java/net/es/oscars/sb/SouthboundPeriodicSyncer.java

```diff
@@ -17,99 +21,191 @@
+
+            // if this WAS a dry run, we are done - we know no changes happened remotely
+            if (dryRynOnly) {
+                return;
+            }
 
-                // now i want to figure out what bits actually synced and what did not
-                nsoVplsStateSyncer.load(vplsPath);
-                nsoVplsStateSyncer.setLocalState(desiredState.getVplsState());
+            // if this wasn't a dry run, we will use the sync results
+            // to set the deployment state for the affected connections
 
-                Enumeration<NsoStateWrapper<NsoVPLS>> enumeration = nsoVplsStateSyncer.getLocalState().elements();
-                while (enumeration.hasMoreElements()) {
-                    NsoStateWrapper<NsoVPLS> wrappedNsoVPLS = enumeration.nextElement();
-                    // This should automatically mark this VPLS as "noop", "add", "delete", or "redeploy"
-                    nsoVplsStateSyncer.evaluate(wrappedNsoVPLS.getInstance().getVcId());
-                }
+            this.updateConnectionStates(vplsSyncResults, lspSyncResults);
 
-                // reload state from NSO after sync
-                // FIXME: modify local connection state according to what NSO gives back
 
-            } catch (Exception e) {
-                log.error("error while syncing vpls state", e);
-                //FIXME: handle exceptions more gracefully
-                throw new RuntimeException(e);
-            }
-            connLock.unlock();
+        } catch (Exception e) {
+            log.error("error while syncing vpls state", e);
+            //FIXME: handle exceptions more gracefully
+            throw new RuntimeException(e);
         }
     }
 
     @Data
     @Builder
     public static class DesiredStateResult {
         Dictionary<Integer, NsoStateWrapper<NsoVPLS>> vplsState;
+        Dictionary<Integer, NsoStateWrapper<NsoLSP>> lspState;
 
     }
 
+    public void updateConnectionStates(Dictionary<Integer, Triple<String, NsoStateSyncer.State, Boolean>> vplsSyncResults,
+                                       Dictionary<Integer, Triple<String, NsoStateSyncer.State, Boolean>> lspSyncResults) {
+        log.info("Updating connection states after sync");
+
+        for (Integer vcId : Collections.list(vplsSyncResults.keys())) {
+            Triple<String, NsoStateSyncer.State, Boolean> syncResult = vplsSyncResults.get(vcId);
+            String connectionId = syncResult.getLeft();
+            NsoStateSyncer.State syncState = syncResult.getMiddle();
+            Boolean success = syncResult.getRight();
+            connSvc.findConnection(connectionId).ifPresent(c -> {
+                DeploymentState ds = c.getDeploymentState();
+                // this is a bit complicated logic
+                switch (syncState) {
+                    case DELETE -> {
+                        if (success) {
```

Comment:
if "success" is true, why would the deployment state be marked as failed?

Reply:
good catch, flipped the logic